js/adapt-articleView.js

```diff
@@ -1,508 +1,487 @@
+        this.$el.addClass('is-disabled');
+        return;
+      }
 
-            this.addChildren();
+      this._blockSliderPreRender();
+    },
 
-            this.$el.addClass('article-block-slider-enabled');
+    _blockSliderPreRender: function() {
+      Adapt.wait.for(function(done){
+        this.resolveQueue = done;
+      }.bind(this));
+      this._blockSliderSetupEventListeners();
+    },
 
-            this.delegateEvents();
+    _blockSliderSetupEventListeners: function() {
 
-            this.$el.imageready(_.bind(function() {
-                _.delay(_.bind(this._blockSliderPostRender, this), 500);
-            }, this));
+      this._blockSliderResizeHeight = this._blockSliderResizeHeight.bind(this);
 
-            return this;
-        },
+      this.listenTo(Adapt, {
+        "device:resize": this._onBlockSliderResize,
+        "device:changed": this._onBlockSliderDeviceChanged,
+        "page:scrollTo": this._onBlockSliderPageScrollTo,
+        "page:scrolledTo": this._onBlockSliderPageScrolledTo
+      });
 
-        _blockSliderConfigureVariables: function() {
-            var blocks = this.model.getChildren().models;
-            var totalBlocks = blocks.length;
+      this.listenToOnce(Adapt, "remove", this._onBlockSliderRemove);
+      this.listenToOnce(this.model, "change:_isReady", this._onBlockSliderReady);
 
-            this.model.set("_currentBlock", 0);
-            this.model.set("_totalBlocks", totalBlocks);
+      var duration = this.model.get("_articleBlockSlider")._slideAnimationDuration || 200;
 
-            var itemButtons = [];
+      this._blockSliderHideOthers = debounce(this._blockSliderHideOthers.bind(this), duration);
 
-            for (var i = 0, l = totalBlocks; i < l; i++) {
-                itemButtons.push({
-                    _className: (i === 0 ? "home" : "not-home") + (" i"+i),
-                    _index: i,
-                    _includeNumber: i !== 0,
-                    _title: blocks[i].get('title')
-                });
-            }
+    },
 
-            this.model.set("_itemButtons", itemButtons);
-        },
+    render: function() {
 
-        _blockSliderConfigureControls: function(animate) {
+      if (this.model.isBlockSliderEnabled()) {
 
-            var duration = this.model.get("_articleBlockSlider")._slideAnimationDuration || 200;
+        this._blockSliderRender();
 
-            if (this._disableAnimationOnce) animate = false;
-            if (this._disableAnimations) animate = false;
+      } else AdaptArticleView.prototype.render.call(this);
 
-            var _currentBlock = this.model.get("_currentBlock");
-            var _totalBlocks = this.model.get("_totalBlocks");
+    },
 
-            var $left = this.$el.find("[data-block-slider='left']");
-            var $right = this.$el.find("[data-block-slider='right']");
+    _blockSliderRender: function() {
+      Adapt.trigger(this.constructor.type + 'View:preRender', this);
```

Comment:
Note sure if this PR is ever going to make it in - suspect @oliverfoster intends to do a complete re-write based on recent work in another project... but just to point out anyway this should really be:
```suggestion
      Adapt.trigger(this.constructor.type + 'View:preRender view:render', this);
```

So as to make sure the article's a11y heading renders